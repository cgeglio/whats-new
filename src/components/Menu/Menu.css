.menu {
  display: flex;
  flex-direction: column;
  grid-area: menu;
}

.menu-button {
  background-color: inherit;
  border: 0;
  color: white;
  font-size: 1.7em;
  height: 2em;
  padding-right: .5em;
  text-align: right;
}

.menu-button:hover {
  background-color: #FFC202;
  border-radius: 0 50px 50px 0px;
}
