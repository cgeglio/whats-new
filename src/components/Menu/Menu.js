import React from 'react';
import './Menu.css';
import logo from '../../images/whats-new-logo.png'


const Menu = (props) => (
  <div className="menu">
    <img src={logo} alt='whats new logo' className="logo-image"/>
    {props.labels.map(l => <button className="menu-button" key={l} onClick={(event) => props.updateNews(l)}>{l[0].toUpperCase() + l.slice(1)}</button>)}
  </div>
);

export default Menu;
