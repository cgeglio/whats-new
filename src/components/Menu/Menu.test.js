import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Menu
      labels={["local", "health"]}
      updateNews={jest.fn()}
    />);
  expect(wrapper).toMatchSnapshot();
  });
});
