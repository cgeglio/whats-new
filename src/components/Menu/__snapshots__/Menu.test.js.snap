// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all data passed in correctly 1`] = `
<div
  className="menu"
>
  <img
    alt="whats new logo"
    className="logo-image"
    src="whats-new-logo.png"
  />
  <button
    className="menu-button"
    key="local"
    onClick={[Function]}
  >
    Local
  </button>
  <button
    className="menu-button"
    key="health"
    onClick={[Function]}
  >
    Health
  </button>
</div>
`;
